### DISCUSSION TITLE
How do I autocomplete a function call without parentheses?

### DISCUSSION THREAD

**Discussion by memchr** [clangd/clangd#1740] [Q&A] [answered]
How do I autocomplete a function call **without** parentheses?

I use clangd with vscode, via the clangd/clangd-vscode extension. This question is similar to #1196, but the opposite behaviour is desired..

Thanks for your time.

**Comment by HighCommander4** [comment, marked as answer]
Clangd has some heuristics to try to detect cases where a reference to a function cannot be / is not likely to be a call, and not insert parentheses in such cases, though the heuristics could use improvement (some relevant issues on file for this are #439, #968, #1677).

If you're looking for a way to disable insertion of parentheses even when writing a call (e.g. because you prefer to type the parentheses yourself), such an option is not currently implemented, but there is https://github.com/llvm/llvm-project/issues/63565 on file for it.
